nexus/src/app/background/tasks/region_snapshot_replacement_step.rs

```diff
@@ -337,25 +351,40 @@ impl RegionSnapshotReplacementFindAffected {
                             status.step_records_created_ok.push(s);
                         }
 
-                        region_snapshot_replacement::InsertStepResult::AlreadyHandled { .. } => {
+                        InsertStepResult::AlreadyHandled { .. } => {
                             info!(
                                 log,
                                 "step already exists for volume id";
                                 "request id" => ?request.id,
                                 "volume id" => ?volume.id(),
                             );
                         }
-                    }
+                    },
 
                     Err(e) => {
                         let s = format!("error creating step request: {e}");
-                        error!(
+                        warn!(
                             log,
                             "{s}";
                             "request id" => ?request.id,
                             "volume id" => ?volume.id(),
                         );
-                        status.errors.push(s);
+
```

Comment:
> the region-snapshot-replace-finish task will call in_progress_region_snapshot_replacement_steps and find there are no replacement steps being resolved; this allows the finish task to try to mark the replacement as having Completed

The only way a region snapshot request transitions out of the Running state (via the finish saga) is if

1) there are no in-progress region snapshot replacement steps, and
2) the request's associated region snapshot record was deleted.

There shouldn't be a way to miss volumes: if there are any volumes referencing the region snapshot, then the record will not have been deleted. In the case that (I think) Alan's referring to, we don't create a region snapshot replacement step for a volume due to an existing lock, but the request will not be finished because of #2 above.

As well, if the request is in the Running state, then the snapshot volume's already had a successful replacement performed, meaning no new volumes that reference the region snapshot can be created, meaning there can't be a race between something creating a new reference (by copying the snapshot volume as a read-only parent) and deleting the last reference.

Reply:
I misread the comment on line 107 of region_snapshot_replacement_finish.rs (it refers to what happens if the conditional branch on line 131 of that file is taken and not to what *has* happened by passing the condition on line 106). This makes more sense now!